File: apps/strapi/scripts/seed.py
#!/usr/bin/env python3
"""Seed script for Strapi using admin authentication"""

import requests
import sys
import json

API_URL = "http://localhost:1337"
ADMIN_EMAIL = "admin@strapi.local"
ADMIN_PASSWORD = "Admin123!"

def main():
    print("🌱 Starting Strapi seed with authentication...")
    
    # Login and get JWT token
    print("🔐 Authenticating...")
    try:
        login_response = requests.post(
            f"{API_URL}/admin/login",
            json={"email": ADMIN_EMAIL, "password": ADMIN_PASSWORD}
        )
        login_response.raise_for_status()
        token = login_response.json()["data"]["token"]
        print("✓ Authenticated")
    except Exception as e:
        print(f"❌ Failed to authenticate: {e}")
        sys.exit(1)
    
    headers = {
        "Content-Type": "application/json",
        "Authorization": f"Bearer {token}"
    }
    
    def post_data(endpoint, name, data):
        try:
            response = requests.post(
                f"{API_URL}/api/{endpoint}",
                json={"data": data},
                headers=headers
            )
            response.raise_for_status()
            result = response.json()
            if "data" in result and "id" in result["data"]:
                print(f"  ✓ {name}")
                return result["data"]["id"]
            else:
                print(f"  ✗ {name}: {result}")
                return None
        except Exception as e:
            print(f"  ✗ {name}: {e}")
            return None
    
    # Seed Genres
    print("📦 Seeding genres...")
    genres_data = [
        {"name": "RPG", "slug": "rpg"},
        {"name": "Visual Novel", "slug": "visual-novel"},
        {"name": "Dating Sim", "slug": "dating-sim"},
        {"name": "Sandbox", "slug": "sandbox"},
        {"name": "Management", "slug": "management"},
    ]
    for genre in genres_data:
        post_data("genres", genre["name"], genre)
    
    # Seed Tags
    print("📦 Seeding tags...")
    tags_data = [
        {"name": "Incest", "slug": "incest", "is_explicit": True},
        {"name": "NTR", "slug": "ntr", "is_explicit": True},
        {"name": "Corruption", "slug": "corruption", "is_explicit": True},
        {"name": "MILF", "slug": "milf", "is_explicit": True},
        {"name": "Male Protagonist", "slug": "male-protagonist", "is_explicit": False},
    ]
    for tag in tags_data:
        post_data("tags", tag["name"], tag)
    
    # Seed Engines
    print("📦 Seeding engines...")
    engines_data = [
        {"name": "Ren'Py", "slug": "renpy"},
        {"name": "Unity", "slug": "unity"},
        {"name": "RPG Maker", "slug": "rpg-maker"},
        {"name": "Unreal Engine", "slug": "unreal-engine"},
    ]
    for engine in engines_data:
        post_data("engines", engine["name"], engine)
    
    # Seed Platforms
    print("📦 Seeding platforms...")
    platforms_data = [
        {"name": "Windows", "slug": "windows"},
        {"name": "Mac", "slug": "mac"},
        {"name": "Linux", "slug": "linux"},
        {"name": "Android", "slug": "android"},
    ]
    for platform in platforms_data:
        post_data("platforms", platform["name"], platform)
    
    # Seed Developer
    print("📦 Seeding developer...")
    dev_data = {
        "name": "Notum Studio",
        "slug": "notum-studio",
        "patreon_link": "https://patreon.com/notumstudio",
        "website_link": "https://notum.cz",
        "twitter": "https://twitter.com/notumcz",
        "discord": "https://discord.gg/notum",
        "subscribestar": "https://subscribestar.com/notum",
    }
    dev_id = post_data("developers", "Notum Studio", dev_data)
    
    if dev_id:
        # Seed Sample Game
        print("📦 Seeding sample game...")
        game_data = {
            "title": "Sample Game",
            "slug": "sample-game",
            "version": "1.0.0",
            "status": "Ongoing",
            "release_date": "2025-01-29",
            "is_featured": True,
            "trending_score": 100,
            "description": "This is a sample game to demonstrate the architecture.",
            "developer": dev_id,
        }
        post_data("games", "Sample Game", game_data)
    
    print("✅ Seed completed!")

if __name__ == "__main__":
    main()

File: apps/strapi/scripts/seed.js
/**
 * Seed script for minimal Strapi data
 * Usage: node scripts/seed.js
 */

const apiUrl = 'http://localhost:1337';

async function seedData() {
  try {
    console.log('🌱 Starting Strapi seed...');

    // Create auth token (for development, we'll skip auth or use a simple approach)
    // In production, you'd need proper authentication

    // Genres
    const genres = [
      { name: 'RPG', slug: 'rpg' },
      { name: 'Visual Novel', slug: 'visual-novel' },
      { name: 'Dating Sim', slug: 'dating-sim' },
      { name: 'Sandbox', slug: 'sandbox' },
      { name: 'Management', slug: 'management' },
    ];

    // Tags
    const tags = [
      { name: 'Incest', slug: 'incest', is_explicit: true },
      { name: 'NTR', slug: 'ntr', is_explicit: true },
      { name: 'Corruption', slug: 'corruption', is_explicit: true },
      { name: 'MILF', slug: 'milf', is_explicit: true },
      { name: 'Male Protagonist', slug: 'male-protagonist', is_explicit: false },
    ];

    // Engines
    const engines = [
      { name: 'Ren\'Py', slug: 'renpy' },
      { name: 'Unity', slug: 'unity' },
      { name: 'RPG Maker', slug: 'rpg-maker' },
      { name: 'Unreal Engine', slug: 'unreal-engine' },
    ];

    // Platforms
    const platforms = [
      { name: 'Windows', slug: 'windows' },
      { name: 'Mac', slug: 'mac' },
      { name: 'Linux', slug: 'linux' },
      { name: 'Android', slug: 'android' },
    ];

    // Seed function
    async function seedCollection(collectionName, data) {
      console.log(`📦 Seeding ${collectionName}...`);
      for (const item of data) {
        try {
          const res = await fetch(`${apiUrl}/api/${collectionName}`, {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({ data: item }),
          });
          if (res.ok) {
            console.log(`  ✓ ${item.name || item.label}`);
          } else {
            const text = await res.text();
            console.log(`  ✗ ${item.name || item.label}: ${res.status} - ${text.substring(0, 100)}`);
          }
        } catch (e) {
          console.log(`  ✗ ${item.name || item.label}: ${e.message}`);
        }
      }
    }

    await seedCollection('genres', genres);
    await seedCollection('tags', tags);
    await seedCollection('engines', engines);
    await seedCollection('platforms', platforms);

    // Seed a sample developer
    console.log('📦 Seeding developer...');
    const developerRes = await fetch(`${apiUrl}/api/developers`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({
        data: {
          name: 'Notum Studio',
          slug: 'notum-studio',
          patreon_link: 'https://patreon.com/notumstudio',
          website_link: 'https://notum.cz',
          twitter: 'https://twitter.com/notumcz',
          discord: 'https://discord.gg/notum',
          subscribestar: 'https://subscribestar.com/notum',
        },
      }),
    });
    if (developerRes.ok) {
      const dev = await developerRes.json();
      console.log(`  ✓ Notum Studio (ID: ${dev.data.id})`);

      // Seed a sample game
      console.log('📦 Seeding sample game...');
      const gameRes = await fetch(`${apiUrl}/api/games`, {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({
          data: {
            title: 'Sample Game',
            slug: 'sample-game',
            version: '1.0.0',
            status: 'Ongoing',
            release_date: new Date().toISOString().split('T')[0],
            is_featured: true,
            trending_score: 100,
            description: 'This is a sample game to demonstrate the architecture.',
            developer: dev.data.id,
            genres: [],
            tags: [],
            platforms: [],
            engine: null,
          },
        }),
      });
      if (gameRes.ok) {
        console.log('  ✓ Sample Game');
      } else {
        const err = await gameRes.json();
        console.log(`  ✗ Sample Game: ${err.error?.message || gameRes.statusText}`);
      }
    } else {
      const err = await developerRes.json();
      console.log(`  ✗ Notum Studio: ${err.error?.message || developerRes.statusText}`);
    }

    console.log('✅ Seed completed!');
  } catch (err) {
    console.error('❌ Seed failed:', err.message);
    process.exit(1);
  }
}

seedData();
